src/overlay/FlowControl.cpp

```diff
@@ -108,15 +123,14 @@ void
 FlowControl::maybeReleaseCapacityAndTriggerSend(StellarMessage const& msg)
 {
     ZoneScoped;
+    releaseAssert(threadIsMain());
 
     if (msg.type() == SEND_MORE || msg.type() == SEND_MORE_EXTENDED)
     {
         if (mNoOutboundCapacity)
         {
-            mApp.getOverlayManager()
-                .getOverlayMetrics()
-                .mConnectionFloodThrottle.Update(mApp.getClock().now() -
-                                                 *mNoOutboundCapacity);
+            mOverlayManager.getOverlayMetrics().mConnectionFloodThrottle.Update(
```

Comment:
I did a bit of refactoring and now we just store OverlayMetrics reference directly inside this class. I didn't use a mutex since OverlayMetrics is just a wrapper around a bunch of medida metrics (the metrics library we use), and those are thread-safe.